test/integration/apiserver/field_validation_test.go

```diff
@@ -2887,6 +2917,103 @@ func testFieldValidationApplyUpdateCRDSchemaless(t *testing.T, rest rest.Interfa
 	}
 }
 
+func testFinalizerValidationApplyCreateAndUpdateCRD(t *testing.T, rest rest.Interface, gvk schema.GroupVersionKind, gvr schema.GroupVersionResource) {
+	var testcases = []struct {
+		name                   string
+		finalizer              string
+		updated_finalizer      string
+		opts                   metav1.PatchOptions
+		strictDecodingError    string
+		strictDecodingWarnings []string
+	}{
+		{
+			name:      "create-crd-with-invalid-finalizer",
+			finalizer: "invalid-finalizer",
+			strictDecodingWarnings: []string{
+				`metadata: Invalid value: "%s": prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers`,
```

Comment:
prefer inlining the exact text we'll get

```suggestion
				`metadata.finalizers: "invalid-finalizer": prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers`,
```

Reply:
you can construct the warning directly like `fmt.Sprintf("metadata.finalizers: %q: prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers", value)` rather than using a field error

Reply:
"Invalid value"  is not defined by this pr. It was defined in https://github.com/kubernetes/kubernetes/blob/8290c4c9c1904f160b8d297ffc052ce58222bc91/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go#L152-L165
So I have to add "Invalid value"  to `expectCreateWarnings`

Reply:
> you can construct the warning directly like `fmt.Sprintf("%s: %q: prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers", fldPath.String(), value)` rather than using a field error

I've got what you means. I'll update this pr later.